base/uk.ac.stfc.isis.ibex.scriptgenerator/src/uk/ac/stfc/isis/ibex/scriptgenerator/PythonBuilder.java

```diff
@@ -54,26 +270,67 @@ public void setRows(Collection<Row> rows) {
 		this.rows = rows;
 	}
 	
+    /**
+     * Sets the SANS settings used in the script.
+     * 
+     * @param settings
+     *            The settings
+     */
+    public void setSettings(SansSettings settings) {
+		this.settings = settings;
+	}
+	
 	/**
 	 * Returns the script back to the view to be displayed.
 	 * @return the completed script
 	 */
 	public String getScript() {
-        return createScript();
+		return script.toString();
 	}
 	
 	/**
-     * Creates a valid Python script.
+     * Calls internal build script methods and adds these returned strings to
+     * the main script.
      * 
-     * Currently does not create a valid script!
+     * @return The Script.
      */
-    private String createScript() {
-        StringBuilder sb = new StringBuilder();
+    public String createScript() {
+		script = new StringBuilder();
```

Comment:
Assuming getScript() can be removed, script no longer needs to be a class variable and can be made local